File: SPOJ/SPOJ_RKS.rb
arr = []
hash=Hash.new(0)
t = gets.split.map(&:to_i)
c = gets.split.map(&:to_i).tally

# TODO MAYBE??
# delete keys higher than t[1]
c.values.each{|el| arr.append(el)}

arr = arr.sort.reverse!
(0..arr.size-1).each{|i|
    tmp = c.key(arr[i])
    arr[i].times {print "#{tmp} "}
    c.delete(tmp)
}

File: SPOJ/SPOJ_RKS.py
from collections import defaultdict


t = list(map(int, input().split()))
c = input().split()

frequency = []
h = defaultdict(lambda: int(0))


def get_key(val):
    for key, value in h.items():
        if val == value:
            return key


for i in range(t[0]):
    h[str(c[i])] += 1
frequency = list(h.values())

frequency.sort(reverse=True)
for i in range(len(frequency)):
    tmp = get_key(frequency[i])
    for j in range(frequency[i]):
        print(tmp, end=' ')
    del h[tmp]
